tsconfig.json

```diff
@@ -0,0 +1,33 @@
+{
+    "compilerOptions": {
+      "target": "ES2018",
+      "module": "ESNext",
+      "moduleResolution": "Node",
+      "lib": [
+        "ESNext",
+        "ESNext.AsyncIterable",
+        "DOM"
+      ],
+      "esModuleInterop": true,
+      "allowJs": true,
+      "sourceMap": true,
+      "strict": true,
+      "noEmit": true,
+      "baseUrl": ".",
+      "paths": {
+        "~/*": [
+          "./*"
+        ],
+        "@/*": [
+          "./*"
+        ]
+      },
+      "types": [
+        "@types/node",
```

Comment:
Help me here to understand, Why do we need to write a custom TS config? Doesn't Nuxt come with an inbuilt TS config?

Reply:
Hey @SagarBajpai, Our initial app was created using the default `javascript` template and the initial inbuilt structure comes only with a `nuxt.config.js` file.
To setup `typescript` in our app, we must create a new file `tsconfig.json` and add all the required types mentioned in the official docs here at : https://typescript.nuxtjs.org/guide/setup/

![image](https://user-images.githubusercontent.com/52736997/150743037-3ded36a4-449c-4091-8140-afe8c6bdd6ff.png)